src/pal/inc/unixasmmacrosarm.inc

```diff
@@ -146,17 +146,46 @@ C_FUNC(\Name\()_End):
 
         __PWTB_StackAlloc = __PWTB_TransitionBlock
 
+	// Make the stack operation compatible with GDB and LIBUNWIND
+	push {r7, lr}
+	mov r7, sp
+	sub sp, #((__PWTB_StackAlloc) + 4 * 4 + 9 * 4)
+
         .ifnc \pushArgRegs, DoNotPushArgRegs
-                PUSH_ARGUMENT_REGISTERS
+		sub r7, #12
+		str r0, [r7]
+		str r1, [r7 , #4]
+		str r2, [r7 , #8]
+		str r3, [r7 , #12]
         .endif
 
-        PUSH_CALLEE_SAVED_REGISTERS
-        
-        alloc_stack     __PWTB_StackAlloc
+	sub r7, #36
+	str r4, [r7]
```

Comment:
Uh.. I'm not supposed to use R12 in this macro because the user of macro (ThePreStub) uses R12 as an argument, "pMethodDesc". I'll stick to what it is except for the modifications in R7.